ANDR-27 Адаптировать топ-бар чтобы не занимал сверху много пространства

Gaileks opened the pull request:
🧩 Что сделано:
• Адаптирован топ-бар (чтобы не занимал сверху много пространства при Landscape
 ориентации ).
• Изменения внесены в функцию TopAppBarWithBottomBorder, DetailQuestionScreenView
🗂 Затронутые модули:
• feature -> detail_question.impl
• feature -> public_questions.impl
• feature -> public_collections.impl
• core -> core_ui
🎯 Цель задачи:
• Уменьшить размер топ-бар при Landscape ориентации на экранах приложения, чтобы не занимал сверху много пространства
<img width="913" height="360" alt="image" src="https://github.com/user-attachments/assets/81e7ed7c-77c2-49bb-a261-42a5b17f7249" />

Gaileks committed:
Адаптировать топ-бар чтобы не занимал сверху много пространства

kish-dev commented on feature/public-collections/impl/src/main/java/ru/yeahub/public_collections/impl/ui/PublicCollectionsScreen.kt:
```diff
@@ -387,42 +393,96 @@ fun TopAppBarWithBottomBorder(
-                    end = Offset(size.width, y),
-                    strokeWidth = borderThicknessPx
+    val configuration = LocalConfiguration.current
+    val isLandscape = configuration.orientation == Configuration.ORIENTATION_LANDSCAPE
```
повторяется код, кажется, это стоит вынести в сам компонент CenterAlignedTopAppBar
Чтобы он сам выбирал как ему рисоваться, в зависимости от режима просмотра

kish-dev commented on core/ui/src/main/java/ru/yeahub/core_ui/component/TopAppBarWithBottomBorder.kt:
```diff
@@ -33,42 +42,93 @@ fun TopAppBarWithBottomBorder(
+    val isLandscape = configuration.orientation == Configuration.ORIENTATION_LANDSCAPE
+
+    if (isLandscape) {
+        CenterAlignedTopAppBar(
```
И сам if тоже стоит вынести.

Для простоты - можно сделать CenterAlignedTopAppBar приватной, а обертку сделать публичной и обозвать ее по-другому

Gaileks committed:
•	Вынес общую логику TopAppBar в переиспользуемый компонент TopAppBarWithBottomBorder
•	Устранил дублирование кода для портретной и ландшафтной ориентаций
•	Все экраны теперь используют единый подход к созданию TopBar

kish-dev commented on feature/public-questions/impl/src/main/java/ru/yeahub/public_questions/impl/presentation/screen/PublicQuestionsScreen.kt:
```diff
@@ -160,53 +151,107 @@ fun HandlePublicQuestionsCommand(
-            }
-    )
-}
+//@OptIn(ExperimentalMaterial3Api::class)
```
Закомментированный код не пушим, с ним потом сложно понять - что делать, когда его раскомментировать нужно, или когда нужно удалить

kish-dev replied:
Поэтому нужно просто удалить)

Gaileks committed:
Удалил закомментированный текст
